WordPress/Classes/System/WordPressAppDelegate+openURL.swift

```diff
@@ -0,0 +1,132 @@
+import WordPressAuthenticator
+
+/// Custom pattern matcher used for our URL scheme
+private func ~=(pattern: String, value: URL) -> Bool {
+    return value.absoluteString.contains(pattern)
+}
+
+private extension Array where Element == URLQueryItem {
+    func value(of key: String) -> String? {
+        return self.first(where: { $0.name == key })?.value
+    }
+    func intValue(of key: String) -> Int? {
+        guard let value = value(of: key) else {
+            return nil
+        }
+        return Int(value)
+    }
+}
+
+private extension URL {
+    func queryParams() -> [URLQueryItem]? {
```

Comment:
Could this be a computed property instead of a function? And maybe rename `queryItems` to match `URLComponents`?

```
var queryItems: [URLQueryItem]? {
...
}
```

Reply:
That sounds great 😀